content/docs/user-guide/dvc-files-and-directories.md

```diff
@@ -1,8 +1,148 @@
+    Specific commit hash, branch or tag name, etc. (a
+    [Git revision](https://git-scm.com/docs/revisions)) used to import the
+    dependency from.
+  - `rev_lock`: Git commit hash of the external <abbr>DVC repository</abbr> at
+    the time of importing or updating (with `dvc update`) the dependency.
+
+Note that comments can be added to `.dvc` files and `dvc.yaml` using the
+`# comment` syntax.
+
+> `meta` fields and `#` comments are preserved among executions of the
+> `dvc repro` and `dvc commit` commands, but not when a `.dvc` file is
+> overwritten by `dvc add`,`dvc import`, or `dvc import-url`.
+
+## dvc.yaml files
+
+When you add commands to a pipeline with `dvc run`, the `dvc.yaml` file is
+created or updated. Here's a simple example:
+
+```yaml
+stages:
+  stageone:
+    cmd: python cmd.py input.data output.data metrics.json
+    deps:
+      - cmd.py
+      - input.data
+    outs:
+      - output.data
+    metrics:
+      - metrics.json
+  stagetwo:
+    cmd: python ...
+    meta: '2nd stage' # User metadata and comments are supported.
+    deps: ...
+```
+
+`dvc.yaml` files consists of a group of `stages` with names provided explicitly
+by the user with the `--name` (`-n`) option of `dvc run`.
+
+Each stage's contents are similar to individual [`dvc` files](#dvcfiles) but
+they can contain more information in `dvc.yaml` These are the possible following
+fields:
+
+- `cmd`: Executable command defined in this stage
+- `deps` (optional): List of <abbr>dependencies</abbr> for this stage
+- `outs` (optional): List of <abbr>outputs</abbr> for this stage
+- `params` (optional): List of the [parameter](/doc/command-reference/params)
+  names and their current values
+- `metrics` (optional): List of [metrics](/doc/command-reference/metrics)
+- `plots` (optional): List of [plot metrics](/doc/command-reference/plots)
+- `frozen` (optional): Whether or not this stage is frozen from reproduction
+- `always_changed` (optional): Whether or not this stage is considered as
+  changed by commands such as `dvc status` and `dvc repro`. `false` by default
+- `meta` (optional): Arbitrary information can be added here manually. Any YAML
+  contents can be added. `meta` contents are ignored by DVC, but they can be
+  useful for user processes that read `.dvc` files.
+
+> `meta` fields and `#` comments are always preserved in `dvc.yaml` stages.
+
+An output entry (`outs`, `metrics`, or `plots`) consists of these fields:
+
+- `md5`: Hash value for the output file
+- `path`: Path to the output in the <abbr>workspace</abbr>, relative to the
+  location of the `.dvc` file
+- `cache` (optional): Whether or not DVC should cache the output. `true` by
+  default
+
+A `dvc.yaml` dependency entry consists of a these possible fields:
+
+- `path`: Path to the dependency, relative to the `wdir` path (always present)
+- `md5`: MD5 hash for the dependency (most [stages](/doc/command-reference/run))
```

Comment:
same here - we don't store md5 in the `dvc.yaml` anymore

Reply:
I've extracted dvc.lock to #1422